modules/win_scheduled_tasks/files/at_task_user_logon.ps1

```diff
@@ -0,0 +1,99 @@
+        [string] $logName = 'Application'
+    )
+    if (!([Diagnostics.EventLog]::Exists($logName)) -or !([Diagnostics.EventLog]::SourceExists($source))) {
+        New-EventLog -LogName $logName -Source $source
+    }
+    switch ($severity) {
+        'DEBUG' {
+            $entryType = 'SuccessAudit'
+            $eventId = 2
+            break
+        }
+        'WARN' {
+            $entryType = 'Warning'
+            $eventId = 3
+            break
+        }
+        'ERROR' {
+            $entryType = 'Error'
+            $eventId = 4
+            break
+        }
+        default {
+            $entryType = 'Information'
+            $eventId = 1
+            break
+        }
+    }
+    Write-EventLog -LogName $logName -Source $source -EntryType $entryType -Category 0 -EventID $eventId -Message $message
+    if ([Environment]::UserInteractive) {
+        $fc = @{ 'Information' = 'White'; 'Error' = 'Red'; 'Warning' = 'DarkYellow'; 'SuccessAudit' = 'DarkGray' }[$entryType]
+        Write-Host -object $message -ForegroundColor $fc
+    }
+}
+
+## Set variable for windows OS
+# Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# Custom facts based off OS details that are not included in the default facts
+
+# Windows release ID.
+# From time to time we need to have the different releases of the same OS version
+$release_key = (Get-ItemProperty -Path 'HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion')
+$release_id = $release_key.ReleaseId
+$win_os_build = [System.Environment]::OSVersion.Version.build
+
+# OS caption
+# Used to determine which KMS license for cloud workers
+$caption = ((Get-WmiObject Win32_OperatingSystem).caption)
+$caption = $caption.ToLower()
+$os_caption = $caption -replace ' ', '_'
+
+if ($os_caption -like "*windows_10*") {
+    $os_version = ( -join ( "win_10_", $release_id))
+}
+elseif ($os_caption -like "*windows_11*") {
+    $os_version = ( -join ( "win_11_", $release_id))
+}
+elseif ($os_caption -like "*2012*") {
+    $os_version = "win_2012"
+}
+else {
+    $os_version = $null
+}
+
+## Accessibilty keys in HKCU
+$Accessibility = Get-ItemProperty -Path "HKCU:\Control Panel\Accessibility"
+
+## Show scrollbars permanently
+switch ($os_version) {
+    "win_11_2009" {
+        ## If it's not there already, create it
+        if ($null -eq $Accessibility.DynamicScrollbars) {
+            Try {
+                New-ItemProperty -Path "HKCU:\Control Panel\Accessibility" -Name "DynamicScrollbars" -Value 0 -ErrorAction Stop
+                Write-Log -message  ('{0} :: Scrollbars successfully set to always show' -f $($MyInvocation.MyCommand.Name)) -severity 'DEBUG'
+            }
+            Catch {
+                Write-Log -message  ('{0} :: Scrollbars unsuccessfully set to always show' -f $($MyInvocation.MyCommand.Name)) -severity 'DEBUG'
```

Comment:
Because this runs at boot and the try/catch will run several times before `HCCU` is available, the log messages should be moved till after the continue. This will reduce thousands of log messages going to Papertrail.